docs/design/pod_evication_and_replacement.md

```diff
@@ -0,0 +1,123 @@
+  - It has been completely drained AND
+  - It is no longer the shard master for any shard
+- DBServer pods can be replaced with another dbserver pod with a different ID on a different node when:
+  - It is not the shard master for any shard OR
+  - For every shard it is the master for, there is an in-sync follower
+- `node.kubernetes.io/unreachable:NoExecute` toleration time is set high to "wait it out a while" (5min)
+- `node.kubernetes.io/not-ready:NoExecute` toleration time is set high to "wait it out a while" (5min)
+
+### Agent Pods
+
+Agent pods run an ArangoDB dbserver as part of an ArangoDB agency.
+It has persistent state potentially tight to the node it runs on and it has a unique ID.
+
+- Agent pods can be evicted from any node as soon as:
+  - It is no longer the agency leader AND
+  - There is at least an agency leader that is responding AND
+  - There is at least an agency follower that is responding
+- Agent pods can be replaced with another agent pod with the same ID but whiped persistent state on a different node when:
+  - The old pod is known to be deleted (e.g. explicit eviction)
+- `node.kubernetes.io/unreachable:NoExecute` toleration time is not set to "wait it out forever"
+- `node.kubernetes.io/not-ready:NoExecute` toleration time is not set "wait it out forever"
+
+### Single Server Pods
+
+Single server pods run an ArangoDB server as part of an ArangoDB single server deployment.
+It has persistent state potentially tight to the node.
+
+- Single server pods cannot be evicted from any node.
+- Single server pods cannot be replaced with another pod.
+- `node.kubernetes.io/unreachable:NoExecute` toleration time is not set to "wait it out forever"
+- `node.kubernetes.io/not-ready:NoExecute` toleration time is not set "wait it out forever"
+
+### Single Pods in Active Failover Deployment
+
+Single pods run an ArangoDB single server as part of an ArangoDB active failover deployment.
+It has persistent state potentially tight to the node it runs on and it has a unique ID.
+
+- Single pods can be evicted from any node as soon as:
+  - It is a follower of an active-failover deployment (Q: can we trigger this failover to another server?)
+- Single pods can always be replaced with another single pod with a different ID on a different node.
+- `node.kubernetes.io/unreachable:NoExecute` toleration time is set high to "wait it out a while" (5min)
+- `node.kubernetes.io/not-ready:NoExecute` toleration time is set high to "wait it out a while" (5min)
+
+### SyncMaster Pods
+
+SyncMaster pods run an ArangoSync as master as part of an ArangoDB DC2DC cluster.
+They have no persistent state, but do have a unique address.
+
+- SyncMaster pods can always be evicted from any node
+- SyncMaster pods can always be replaced with another syncmaster pod on a different node
+- `node.kubernetes.io/unreachable:NoExecute` toleration time is set low (15sec)
+- `node.kubernetes.io/not-ready:NoExecute` toleration time is set low (15sec)
```

Comment:
Is there any requirement about the same network endpoint or an internal k8s service being set up in case of a replacement?

Reply:
no